vj4/handler/record.py

```diff
@@ -138,17 +141,16 @@ class RecordDetailHandler(base.Handler, RecordMixin):
       return
     show_status = True
     if rdoc['tid']:
-      tdoc = await contest.get(rdoc['domain_id'], rdoc['tid'])
-      show_status = self.tdoc_visible(tdoc)
+      show_status, tdoc = await self.rdoc_contest_visible(rdoc)
     else:
-      tdoc = None
+      show_status, tdoc = False, None
     # TODO(twd2): futher check permission for visibility.
     if (not self.own(rdoc, field='uid')
         and not self.has_perm(builtin.PERM_READ_RECORD_CODE)
         and not self.has_priv(builtin.PRIV_READ_RECORD_CODE)):
       del rdoc['code']
     if not show_status and 'code' not in rdoc:
-      raise error.PermissionError(builtin.PERM_VIEW_CONTEST_HIDDEN_STATUS)
+      raise error.PermissionError(builtin.PERM_VIEW_CONTEST_HIDDEN_STATUS_AND_SCOREBOARD)
```

Comment:
???

Reply:
When holding this permission, the user can view hidden record status, as well as hidden scoreboard.